src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -132,8 +132,32 @@ async def _run(self) -> None:  # pylint: disable=too-many-locals
         """
         await self._component_manager.start()
 
+        power_tasks: dict[str, asyncio.Task[None]] = {}
+
+        def cleanup(task: asyncio.Task[None]) -> None:
+            task_name = task.get_name()
+            assert task_name in power_tasks, "Task not found in power_tasks"
+
+            # Raise if there was an exception while processing the task
+            task.result()
+
+            del power_tasks[task_name]
+
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            task_name = f"power_request_ids={request.component_ids}"
+            if task_name in power_tasks:
+                result = Error(
+                    request=request,
+                    msg="A request for the same components is being processed. Skipped.",
+                )
+                await self._result_sender.send(result)
```

Comment:
I think we shouldn't call this an `Error`.  I guess we need a new result type.  Maybe `Discarded`.

But I would rather not send a message at all,  and keep the latest request and discard just the older ones, and execute the latest request as soon as the running one is complete.

Reply:
This is not longer sending the Error, instead it is keeping track of the latest received power request

Reply:
@shsms your input is needed here?

My understanding is it's only needed to send a response if the request is processed (either succeeded, partial failed, error, etc). The current patch works as a request replacement if the previous request was queued up

Reply:
Yes.  Ever since the power manager was introduced, there has not been a one-to-one mapping between user proposals and response from the power distributor.  So just a log message should be enough saying new request received before previous one could start executing, so it is discarded.  And I think we have something of this form.